rflx/specification/parser.py

```diff
@@ -565,13 +565,13 @@ def create_function_decl(
             arguments.append(
                 decl.Argument(
                     create_id(p.f_identifier, filename),
-                    create_id(p.f_type_identifier, filename),
+                    model.qualified_type_identifier(create_id(p.f_type_identifier, filename)),
                 )
             )
     return decl.FunctionDeclaration(
         create_id(declaration.f_identifier, filename),
         arguments,
-        create_id(declaration.f_return_type_identifier, filename),
+        model.qualified_type_identifier(create_id(declaration.f_return_type_identifier, filename)),
         location=node_location(declaration, filename),
     )
```

Comment:
IMHO there should be at least a unit test for this behavior change. For example:

```diff
diff --git a/tests/unit/specification/grammar_test.py b/tests/unit/specification/grammar_test.py
index 7a00d944..d2b77fc2 100644
--- a/tests/unit/specification/grammar_test.py
+++ b/tests/unit/specification/grammar_test.py
@@ -574,6 +574,14 @@ def test_channel_declaration(string: str, expected: decl.Declaration) -> None:
             "with function X (A : B; C : D) return Y",
             decl.FunctionDeclaration("X", [decl.Argument("A", "B"), decl.Argument("C", "D")], "Y"),
         ),
+        (
+            "with function X (A : Boolean) return Boolean",
+            decl.FunctionDeclaration(
+                "X",
+                [decl.Argument("A", str(model.BOOLEAN.identifier))],
+                str(model.BOOLEAN.identifier),
+            ),
+        ),
     ],
 )
 def test_formal_function_declaration(string: str, expected: decl.Declaration) -> None:
```

Reply:
I agree and thanks for providing it :-) Fixed in 4f26db5b9bc56573ccf0b3faeb38317b70fa9d44